File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;
//[1,5,3,5,1,2,1,4]
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> basket = new Stack<>();
        int pick;
        for(int i : moves){
            for(int j=0; j<board.length; j++){
                pick = board[j][i-1];
                if(pick!=0){
                    if(!basket.empty() && basket.peek()==pick){
                        answer += 2;
                        basket.pop();
                    }else{
                        basket.push(pick);
                    }
                    board[j][i-1] = 0;
                    break;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var answer = 0;
    var arr = [];
    moves.forEach((col) => {
        for(var row=0; row<board.length; row++) {
            var n = board[row][col-1]
            if(n) {
                board[row][col-1] = 0;
                if(arr[arr.length-1] === n) {
                    arr.pop();
                    answer += 2;
                }
                else arr.push(n);
                break;
            }
        }
    })
    return answer;
}